pkg/monitors/kubernetes/kubeletmetrics/monitor.go

```diff
@@ -0,0 +1,167 @@
+	// `/stats/summary` endpoint.
+	UsePodsEndpoint bool `yaml:"usePodsEndpoint"`
+}
+
+// Monitor for K8s volume metrics as reported by kubelet
+type Monitor struct {
+	Output        types.Output
+	cancel        func()
+	kubeletClient *kubelet.Client
+	dimCache      map[string]map[string]string
+}
+
+// Configure the monitor and kick off volume metric syncing
+func (m *Monitor) Configure(conf *Config) error {
+	logger := log.WithFields(log.Fields{"monitorType": monitorType})
+
+	var err error
+	m.kubeletClient, err = kubelet.NewClient(&conf.KubeletAPI)
+	if err != nil {
+		return err
+	}
+
+	var ctx context.Context
+	ctx, m.cancel = context.WithCancel(context.Background())
+	utils.RunOnInterval(ctx, func() {
+		logger.Debug("Collecting kubelet /stats/summary metrics")
+		dps, err := m.getSummaryMetrics(conf.UsePodsEndpoint)
+		if err != nil {
+			logger.WithError(err).Error("Could not get summary metrics")
+			return
+		}
+		logger.Debugf("Sending kubelet metrics: %v", dps)
+
+		m.Output.SendDatapoints(dps...)
+	}, time.Duration(conf.IntervalSeconds)*time.Second)
+
+	return nil
+}
+
+func (m *Monitor) getSummaryMetrics(usePodsEndpoint bool) ([]*datapoint.Datapoint, error) {
+	var podsByUID map[k8stypes.UID]*v1.Pod
+	if usePodsEndpoint {
+		var err error
+		podsByUID, err = m.getPodsByUID()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	summary, err := m.getSummary()
+	if err != nil {
+		return nil, err
+	}
+
+	var dps []*datapoint.Datapoint
+
+	for _, p := range summary.Pods {
+		podResource := podsByUID[k8stypes.UID(p.PodRef.UID)]
+		if usePodsEndpoint && podResource == nil {
+			continue
+		}
+
+		dims := map[string]string{
+			"kubernetes_pod_uid":   p.PodRef.UID,
+			"kubernetes_pod_name":  p.PodRef.Name,
+			"kubernetes_namespace": p.PodRef.Namespace,
+		}
+
+		if p.EphemeralStorage != nil {
+			if p.EphemeralStorage.CapacityBytes != nil {
+				dps = append(dps, sfxclient.Gauge(podEphemeralStorageCapacityBytes, dims, int64(*p.EphemeralStorage.CapacityBytes)))
+			}
+			if p.EphemeralStorage.UsedBytes != nil {
+				dps = append(dps, sfxclient.Gauge(podEphemeralStorageUsedBytes, dims, int64(*p.EphemeralStorage.UsedBytes)))
+			}
+		}
+
+		if p.Network != nil && p.Network.Interfaces != nil {
+			for _, i := range p.Network.Interfaces {
+				intfDims := utils.CloneStringMap(dims)
+
+				intfDims["interface"] = i.Name
+
+				if i.RxBytes != nil {
+					dps = append(dps, sfxclient.Cumulative(podNetworkReceiveBytesTotal, intfDims, int64(*i.RxBytes)))
+				}
+				if i.TxBytes != nil {
+					dps = append(dps, sfxclient.Cumulative(podNetworkTransmitBytesTotal, intfDims, int64(*i.TxBytes)))
+				}
+				if i.RxErrors != nil {
+					dps = append(dps, sfxclient.Cumulative(podNetworkReceiveErrorsTotal, intfDims, int64(*i.RxErrors)))
+				}
+				if i.TxErrors != nil {
+					dps = append(dps, sfxclient.Cumulative(podNetworkTransmitErrorsTotal, intfDims, int64(*i.TxErrors)))
+				}
+			}
+		}
+
```

Comment:
Do you think pod memory and pod cpu metrics could be worth sending?

Reply:
Ditto for node-level metrics (cpu, fs, memory, network), or is that expected to be sent via another monitor?

Reply:
My understanding is that pod cpu/memory metrics are just aggregates of the individual container metrics and thus would be redundant -- you could just sum up the containers to get the pod.  Have you seen anything that says differently?

Reply:
> Ditto for node-level metrics (cpu, fs, memory, network), or is that expected to be sent via another monitor?

Those are all sent by the host monitors (e.g. `memory`, `cpu`, etc.).